src/uthash.h

```diff
@@ -983,14 +1058,26 @@ do {
           }                                                                      \
           if ((dst) == NULL) {                                                   \
             DECLTYPE_ASSIGN(dst, _elt);                                          \
-            HASH_MAKE_TABLE(hh_dst, dst);                                        \
+            HASH_MAKE_TABLE(hh_dst, dst, _mem_ok);                               \
+            if (HASH_NOMEM_OK && !_mem_ok) {                                     \
+              uthash_fatal(_elt);                                                \
+              dst = NULL;                                                        \
+              continue;                                                          \
+            }                                                                    \
           } else {                                                               \
             _dst_hh->tbl = (dst)->hh_dst.tbl;                                    \
           }                                                                      \
           HASH_TO_BKT(_dst_hh->hashv, _dst_hh->tbl->num_buckets, _dst_bkt);      \
-          HASH_ADD_TO_BKT(_dst_hh->tbl->buckets[_dst_bkt], _dst_hh);             \
-          HASH_BLOOM_ADD(_dst_hh->tbl, _dst_hh->hashv);                          \
+          HASH_ADD_TO_BKT(_dst_hh->tbl->buckets[_dst_bkt], hh_dst, _dst_hh,      \
+              _mem_ok);                                                          \
```

Comment:
Style nit: I'd prefer you not wrap this line; just go ahead and let it extend past the usual margin.

    HASH_TO_BKT(_dst_hh->hashv, _dst_hh->tbl->num_buckets, _dst_bkt);      \
    HASH_ADD_TO_BKT(_dst_hh->tbl->buckets[_dst_bkt], hh_dst, _dst_hh, _mem_ok); \
    (dst)->hh_dst.tbl->num_items++;                                        \